sbndqm/dqmAnalysis/PMT/CAENV1730Streams_module.cc

```diff
@@ -0,0 +1,251 @@
+  else if (threshAlgName == "SlidingWindow")
+    threshAlg = new pmtana::AlgoSlidingWindow(hit_alg_pset);
+  else if (threshAlgName == "FixedWindow")
+    threshAlg = new pmtana::AlgoFixedWindow(hit_alg_pset);
+  else if (threshAlgName == "CFD" )
+    threshAlg = new pmtana::AlgoCFD(hit_alg_pset);
+  else throw art::Exception(art::errors::UnimplementedFeature)
+              << "Cannot find implementation for "
+                    << threshAlgName << " algorithm.\n";
+
+  pulseRecoManager.AddRecoAlgo(threshAlg);
+  
+}
+
+//------------------------------------------------------------------------------------------------------------------
+
+
+sbndaq::CAENV1730Streams::~CAENV1730Streams(){};
+
+//------------------------------------------------------------------------------------------------------------------
+
+
+void sbndaq::CAENV1730Streams::clean() {
+
+  m_get_temperature.clear();
+
+}
+
+
+//-------------------------------------------------------------------------------------------------------------------
+
+int16_t sbndaq::CAENV1730Streams::Median(std::vector<int16_t> data, size_t n_adc) 
+{
+  // First we sort the array
+  std::sort(data.begin(), data.end());
+
+  // check for even case
+  if (n_adc % 2 != 0)
+    return data[n_adc / 2];
+
+  return (data[(n_adc - 1) / 2] + data[n_adc / 2]) / 2.0;
+}
+
+
+double sbndaq::CAENV1730Streams::Median(std::vector<double> data, size_t n_adc) 
+{
+  // First we sort the array
+  std::sort(data.begin(), data.end());
+
+  // check for even case
+  if (n_adc % 2 != 0)
+    return data[n_adc / 2];
+
+  return (data[(n_adc - 1) / 2] + data[n_adc / 2]) / 2.0;
+}
+
+
+//------------------------------------------------------------------------------------------------------------------
+
+
+double sbndaq::CAENV1730Streams::RMS(std::vector<int16_t> data, size_t n_adc, int16_t baseline )
+{
+  double ret = 0;
+  for (size_t i = 0; i < n_adc; i++) {
+    ret += (data[i] - baseline) * (data[i] - baseline);
+  }
+  return sqrt(ret / n_adc);
+}
+
+
+//------------------------------------------------------------------------------------------------------------------
+
+
+int16_t sbndaq::CAENV1730Streams::Min(std::vector<int16_t> data, size_t n_adc, int16_t baseline )
+{
+  int16_t min = baseline;
+  for (size_t i = 0; i < n_adc; i++) {
+    if( data[i] < min ) { min = data[i]; } ;
+  }
+  return min;
+}
+
+//------------------------------------------------------------------------------------------------------------------
+
+void sbndaq::CAENV1730Streams::analyze(art::Event const & evt) {
+
+  // Get the event timestamp
+
+  int level = 3; 
+
+  artdaq::MetricMode mode = artdaq::MetricMode::Average;
+  artdaq::MetricMode rate = artdaq::MetricMode::Rate;
+
+  std::string groupName = "PMT";
+
+
+  // We check the fragments 
+  auto digitizerinfoHandle = evt.getValidHandle< std::vector<pmtAnalysis::PMTDigitizerInfo> >( m_pmtditigitizerinfo_tag );
+
+  if( digitizerinfoHandle->size() > 0 ) {
```

Comment:
While in general I would write data product access like you did, this is a very specific workflow where we must catch all errors and move on.
I assume that the workflow requires the special decoder to be running, in which case `pmtAnalysis::PMTDigitizerInfo` will be always produced; but this may actually not happen if the decoder itself had an error and then did not `put()` the required information. In that case, the original code will throw an exception, which we want to avoid.
```suggestion
  art::Handle<std::vector<pmtAnalysis::PMTDigitizerInfo>> digitizerinfoHandle;
  evt.getByLabel( m_pmtditigitizerinfo_tag, digitizerinfoHandle );
  if( digitizerinfoHandle->isValid() && !digitizerinfoHandle->empty() ) {
```